src/sql/src/plan/statement/ddl.rs

```diff
@@ -347,12 +347,25 @@ pub fn plan_create_source(
 
     let (external_connector, encoding) = match connector {
         CreateSourceConnector::Kafka(kafka) => {
-            let (broker, topic) = match &kafka.connector {
-                mz_sql_parser::ast::KafkaConnector::Inline { broker } => (broker, &kafka.topic),
-                // Temporary until the rest of the connector plumbing is finished
-                mz_sql_parser::ast::KafkaConnector::Reference { .. } => unreachable!(),
+            let (broker, topic, options) = match &kafka.connector {
+                mz_sql_parser::ast::KafkaConnector::Inline { broker } => {
+                    (broker.to_owned(), &kafka.topic, None)
+                }
+                mz_sql_parser::ast::KafkaConnector::Reference { connector } => {
+                    let connector_name = normalize::unresolved_object_name(connector.clone())?;
+                    let item = scx.catalog.resolve_item(&connector_name)?;
+                    let connector = item.catalog_connector()?;
+                    depends_on.push(item.id());
+                    (connector.uri(), &kafka.topic, Some(connector.options()))
+                }
+            };
+
+            // If we got options from a connector then use those, they have already had kafka_util::extract_config run on them
+            let config_options = if let Some(opts) = options {
+                opts
```

Comment:
This looks to me like it's dropping the with options specified with the create source on the ground. If the parser already took care of this and it's an impossible condition, maybe add an assert to ensure we aren't doing it inadvertently?

Reply:
This is a tricky situation since the with_options in create source can be for either or both of the kafka broker and confluent schema registry, and the intent for creating a source with a connector is that you cannot supply broker options in both places, hence if we have a connector we unconditionally use them for the broker. 
The with_options from the create source aren't discarded, they're just not used in the `KafkaSourceConnector` which is only concerned with the broker and not the schema registry. I am confident this functionally works right because in the kafka-ssl smoketest we get broker options from the connector and schema registry options from the statement and everything works as expected.
Once the syntax for creating sources is refactored to eliminate these with_options then this either/or will be exposed in the grammar directly. I was thinking about checking for option collisions between the connector and the `CREATE SOURCE` statement and erroring if anything is duplicated but that will not provide perfect coverage since it's possible that the connector and create source statement would choose non-overlapping options i.e. the `CREATE SOURCE` statement could just specify username/pass and connector could specify security_protocol and SSL certificate options. Both sets are valid on their own, and they have no overlap. The only way to reliably validate would be to check that nothing specified in the `CREATE SOURCE` statement `WITH` options is a valid option for a connector.

Reply:
Ah, the thing that assuages my concern is that we ensure that the options are empty using `normalize::ensure_empty_options`; that prevents someone from defining one of the connector options.